src/redux/books/books.js

```diff
@@ -35,32 +18,58 @@ export const booksReducer = (state = defaultState, action = {}) => {
       }];
 
     case UPDATE: {
-      return state;
+      return [...state, ...action.payload.items];
     }
 
     case DELETE:
       return state.filter((book) => (book.id !== action.id));
 
-    case LOAD: {
-      return state;
-    }
-
     default: {
       return state;
     }
   }
 };
 
 // Action Creators
-export const createBook = (id, title, author) => ({
+export const createBook = (items) => ({
   type: ADD,
-  id,
-  title,
-  author,
+  id: items.id,
+  title: items.title,
+  author: items.author,
 });
 
-export const loadBooks = () => ({ type: LOAD });
-
 export const deleteBook = (id) => ({ type: DELETE, id });
 
-export const updateBook = (book) => ({ type: UPDATE, book });
+export const updateBook = (items) => ({ type: UPDATE, payload: { items } });
+
+// Thunks
+
+export const fetchBooks = () => async (dispatch) => fetch(APILINK)
+  .then((res) => res.json())
+  .then((data) => {
+    const dataInput = [];
+    Object.values(data).forEach((value) => {
+      dataInput.push(value[0]);
+    });
+    Object.keys(data).forEach((key, i) => {
+      dataInput[i].id = key;
+    });
+    dispatch(updateBook(dataInput));
+  });
+
+export const postBook = (items) => async (dispatch) => {
+  fetch(APILINK, {
+    method: 'POST',
+    body: JSON.stringify(items),
+    headers: {
+      'Content-type': 'application/json; charset=UTF-8',
+    },
+  }).then(() => dispatch(createBook(items)));
+};
+
+export const deleteBookItem = (id) => async (dispatch) => {
+  fetch(`${APILINK}/${id}`, {
+    method: 'DELETE',
+    body: JSON.stringify(id),
+  }).then(() => dispatch(deleteBook(id)));
+};
```

Comment:
- [ ] Please create your thunk functions using `createAsyncThunk` instead of creating them normally, as per [this](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/connec_to_api.md#:~:text=Use%20fetch%20or%20axios%20with%20createAsyncThunk%20for%20making%20API%20calls.) requirement.